### PR TITLE
Update FreeBayes to version 1.3.1

### PR DESCRIPTION
**Pull request by davebx**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR updates an existing tool or tool collection

This replaces #2455

### CODE REVIEW

File: tools/freebayes/macros.xml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <macros>
-    <token name="@DEPENDENCY_VERSION@">1.1.0.46</token>
+    <token name="@DEPENDENCY_VERSION@">1.3.1</token>
     <xml name="requirements">
         <requirements>
             <requirement type="package" version="@DEPENDENCY_VERSION@">freebayes</requirement>
```
Comment (wm75) [inline tools/freebayes/macros.xml]: would it be possible to bump also the samtools requirement to 1.9? Not sure, but samtools seems to be used only for ref indexing and getting the input bam header so should be unproblematic

Comment (wm75) [reply tools/freebayes/macros.xml]: also, remove the `-0` from the freebayes and leftalign tool wrapper versions. Just use `@DEPENDENCY_VERSION@` there.

File: tools/freebayes/macros.xml [hunk 2]
```diff
@@ -44,6 +44,8 @@
         </conditional>
     </xml>
     <xml name="par_min_cov">
-        <param name="min_coverage" argument="--coverage" type="integer" value="0" label="Require at least this coverage to process a site" />
+        <param name="min_coverage" argument="--min-coverage" type="integer" value="0" label="Require at least this coverage to process a site" />
+        <param name="limit_coverage" argument="--limit-coverage" type="integer" value="0" label="Downsample per-sample coverage to this level if greater than this coverage" />
+        <param name="skip_coverage" argument="--skip-coverage" type="integer" value="0" label="Skip processing of alignments overlapping positions with coverage greater than this" />
```
Comment (wm75) [inline tools/freebayes/macros.xml]: I don't think this is the best place to add in the new options. The way the wrapper structures things, this ties `--limit-coverage` and `--skip-coverage` to using standard filters unless you are willing to configure all options.
If you do this, then I would also add a boolean for turning on/off standard filters.

**Comment by davebx** [comment]
Thanks for the suggestions, @wm75, I'll be making those changes soon.

File: tools/freebayes/macros.xml (hunk 2 above)
Comment (davebx) [reply tools/freebayes/macros.xml]: I like this suggestion, but I want to implement it in a way that doesn't hinder rerunning jobs, so I'll have to give it a little more thought when I have time (which will be this week, of course)

File: tools/freebayes/macros.xml (hunk 1 above)
Comment (davebx) [reply tools/freebayes/macros.xml]: Done and done, thanks.

File: tools/freebayes/macros.xml (hunk 2 above)
Comment (wm75) [reply tools/freebayes/macros.xml]: Good point! Take your time.

**Comment by bernt-matthias** [comment]
Did you check that all changes mentioned in https://github.com/ekg/freebayes/releases are reflected? In particular the changed defaults in version 1.2.

**Comment by nsoranzo** [comment]
Can you please rebase the pull request? Unfortunately we had to do a force-push on the master branch.

Please let us know if you have issues and we can do it for you.

**Comment by bgruening** [comment]
@bernt-matthias do you think we can merge this?
@davebx do you plan to work on this and adapt the PR?

**Comment by bernt-matthias** [comment]
The defaults that changed in 1.2.0 should be changed: https://github.com/ekg/freebayes/releases/tag/v1.2.0

**Comment by davebx** [comment]
Replaced with #2525

**Pull request closed without merging**